src/NServiceBus.Persistence.DynamoDB/Outbox/OutboxPersister.cs

```diff
@@ -62,23 +63,42 @@ public OutboxPersister(IAmazonDynamoDB dynamoDbClient, OutboxPersistenceConfigur
             bool responseItemsHasOutboxMetadataEntry = false;
             if (foundOutboxMetadataEntry == null && response.Items.Count >= 1)
             {
-                foundOutboxMetadataEntry = true;
-                var headerItem = response.Items[0];
-                // In case the metadata is not marked as dispatched we want to know the number of transport operations
-                // in order to pre-populate the lists etc accordingly
-                if (!headerItem[Dispatched].BOOL)
+                var potentialHeaderItem = response.Items[0];
+                // Batch delete of transport operations could leave phantom records and we might be reading those
+                if (potentialHeaderItem[configuration.Table.SortKeyName].S == outboxMetadataSortKey)
                 {
-                    numberOfTransportOperations = Convert.ToInt32(headerItem[OperationsCount].N);
+                    foundOutboxMetadataEntry = true;
+                    // In case the metadata is not marked as dispatched we want to know the number of transport operations
+                    // in order to pre-populate the lists etc accordingly
+                    if (!potentialHeaderItem[Dispatched].BOOL)
+                    {
+                        numberOfTransportOperations = Convert.ToInt32(potentialHeaderItem[OperationsCount].N);
+                    }
+                    responseItemsHasOutboxMetadataEntry = true;
                 }
-                responseItemsHasOutboxMetadataEntry = true;
+            }
+
+            // the metadata entry needs to be the first element within that partition key range. If it wasn't found
+            // let's skip further evaluation because we would be reading phantom records only.
+            if (!foundOutboxMetadataEntry.GetValueOrDefault(false))
+            {
+                continue;
```

Comment:
shouldn't this be a break if we want to skip further loops?

Reply:
Yeah that makes things simpler. Sorry didn't have time to give this a second look